### ISSUE TITLE
Rule Request: [duplicate_target_in_package]

### ISSUE THREAD

**Issue by skydivedan** [realm/SwiftLint#4805]
### New Issue Checklist

- [ ] Updated SwiftLint to the latest version
- [ ] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### New rule request

This rule would warn about duplicate targets in a `Package.swift` file.

Violation:
```
let package = Package(
    name: "MyPackage",
    platforms: [.iOS(.v14)],
    products: [
        .library(
            name: "MyPackage",
            targets: [
                "BigDependency",
                "SmallDependency",
                "OtherDependency",
                "BigDependency"
            ]),
    ],
```

Non-violation
```
let package = Package(
    name: "MyPackage",
    platforms: [.iOS(.v14)],
    products: [
        .library(
            name: "MyPackage",
            targets: [
                "BigDependency",
                "SmallDependency",
                "OtherDependency"
            ]),
    ],
```

I've seen this problem occur usually when there are two people working on the same swift package. One of them adds a dependency. And the other adds another, and then the branches are merged. It isn't flagged as a problem by the compiler, but can cause issues.
